src/PipeScore/Gracenote/gracenotes.ts

```diff
@@ -183,7 +183,7 @@ gracenotes.set('c-shake', (note, prev) => {
   return invalidateIf(note !== Pitch.D, pitches);
 });
 
-gracenotes.set('toarluath', (note, prev) => {
+gracenotes.set('taorluath', (note, prev) => {
```

Comment:
Thanks very much for the PR! Unfortunately the problem with changing this is that everyone's scores will be saved with the typo. So either we just change the documentation and leave the code/score files with the typo, or add something to `ReactiveGracenote.fromObject()` to update the typo in older versions of the score. I can do this if you want.

Reply:
I see...  If committed as-is, Pipescore will read a score containing a taorluath and complain because it found `'toarluath`, and maybe nasal-daemon time.  The first solution seems like sweeping under the rug.

The solution I was thinking of would be to temporarily shut off the web server daemon, apply the fix, do a search-and-replace on all the scores to correct the typo, and then turn the web server daemon back on.

The second solution, I'm guessing that `ReactiveGracenote.fromObject()` would look for the misspelled token and substitute the correct one.  Then by attrition corrections would be applied.

How about a third way?  It looks like scores are not sanity-checked for invalid tokens after being read in.  Could the misspelling correction by attrition be done there?  Having that sanity check alone would be a very good thing going forward with importing and exporting scores.

Reply:
I'm very wary of fuddling about with people's scores on the backend in case I accidentally destroy data, so I'd prefer option 2. Option 3 seems to be a bit overkill - since the files are only ever made by PipeScore they should always be 'correct' unless the user themselves has edited the file, in which case that's their fault... the worst that happens if an invalid score is read in is that that score doesn't open properly, and adding in a huge amount of validation logic is not really worth it IMO.

I've already done similar things to option 2 when changing the score format, e.g. [here](https://github.com/macarc/PipeScore/blob/3c9c773aeb5b19240516068455051f8420440025/src/PipeScore/TextBox/index.ts#L53C12-L53C12) and [here](https://github.com/macarc/PipeScore/blob/3c9c773aeb5b19240516068455051f8420440025/src/PipeScore/Note/base.ts#L64).

Reply:
How would a user edit their score file without going through PipeScore itself?  My concern is that if this functionality is added AND no validation logic is used, then a user might be able to cause serious security problems.  How big would this validation logic code be?

This might go hand-in-hand with allowing users to export and import scores in PipeScore's own format.  Allowing for that would allow users on their own to move scores from one server to another.  For instance, a band might want to maintain its own installation of PipeScore.  If such functionality is added, a validation check only upon upload shouldn't be a significant drain on the server.  I see how it would be a big drain if such validation is performed each time a user opens a score already on the server.  

That being said, I agree that option 2 would be the best solution for the misspelling.

Reply:
Right now, the backend treats all scores as just arbitrary data (it doesn't do any parsing/etc. of scores). The only place an invalid score could cause issues would be when handling it on the frontend, which means that there isn't really any security issue associated with an invalid score. I think adding a simple check to the `ReactiveGracenote.fromObject()` method is the way to go - I'll accept this PR then add that change on top! Thank you :)